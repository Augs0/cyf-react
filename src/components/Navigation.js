import React from 'react'

import { Link } from 'react-router-dom'

import { Nav, Navbar, NavItem } from 'react-bootstrap'
import { LinkContainer } from 'react-router-bootstrap'

import cyflogo from '../img/cyf_brand.png';
import fblogo from '../img/ico/fb_logo.svg';
import twitterlogo from '../img/ico/twitter-silhouette.svg';
import linkedinlogo from '../img/ico/linkedin-logo.svg';
import emaillogo from '../img/ico/email-icon.svg';

const logoStyle = {
  width: '20px',
  height: '21px'
}

const cyflogoStyle = {
  width: '158px',
  height: '50px'
}


export const Navigation = () => (
  <Navbar collapseOnSelect>
    <Navbar.Header>
      <Link to="/">
        <img src={cyflogo} style={cyflogoStyle} alt="logo" />
      </Link>
      <Navbar.Toggle />
    </Navbar.Header>
    <Navbar.Collapse>
    <Nav pullRight>
      {/*<NavItem>
        <Link to="/events">Events</Link>
      </li>*/}
      <LinkContainer to="/about">
        <NavItem eventKey={1}>About</NavItem>
      </LinkContainer>
      <LinkContainer to="/students">
      <NavItem>Students</NavItem>
    </LinkContainer>
     <LinkContainer to="/volunteers">
      <NavItem>Volunteers</NavItem>
    </LinkContainer>
     <LinkContainer to="/partners">
      <NavItem>Partners</NavItem>
    </LinkContainer>
     <LinkContainer to="/meet-the-team">
      <NavItem>Meet the team</NavItem>
    </LinkContainer>
     <LinkContainer to="https://www.facebook.com/codeyourfuture.co" target="_blank">
     <NavItem className="nav-icon">
          <img src={fblogo} style={logoStyle} alt="facebook"/>
     </NavItem>
    </LinkContainer>
     <LinkContainer to="https://twitter.com/CodeYourFuture_" target="_blank">
     <NavItem className="nav-icon">
          <img src={twitterlogo} style={logoStyle} alt="twitter"/>
     </NavItem>
    </LinkContainer>
     <LinkContainer to="https://www.linkedin.com/company/codeyourfuture" target="_blank">
     <NavItem className="nav-icon">
          <img src={linkedinlogo} style={logoStyle} alt="linkedin"/>
     </NavItem>
    </LinkContainer>
     <LinkContainer to="mailto:contact@codeyourfuture.co" title="Contact us">
     <NavItem className="nav-icon">
          <img src={emaillogo} style={logoStyle} alt="email"/>
     </NavItem>
    </LinkContainer>
    </Nav>
    </Navbar.Collapse>
  </Navbar>
);
