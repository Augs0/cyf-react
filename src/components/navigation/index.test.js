import React from 'react';
import { shallow } from 'enzyme';

import Navigation from './';

describe('Navigation Component', () => {
  it('renders', () => {
    expect(shallow(<Navigation />)).toMatchSnapshot();
  });
});
