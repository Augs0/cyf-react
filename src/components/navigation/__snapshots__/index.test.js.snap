// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  collapseOnSelect={true}
>
  <NavbarHeader>
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="logo"
        src="cyf_brand.png"
        style={
          Object {
            "height": "50px",
            "width": "158px",
          }
        }
      />
    </Link>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      onSelect={[Function]}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/about"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={1}
        >
          About
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/students"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={1}
        >
          Students
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/volunteers"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={1}
        >
          Volunteers
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/partners"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={1}
        >
          Partners
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/meet-the-team"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={1}
        >
          Meet the team
        </NavItem>
      </LinkContainer>
      <NavItem
        active={false}
        className="nav-icon"
        disabled={false}
        eventKey="blog"
      >
        Blog
      </NavItem>
      <NavItem
        active={false}
        className="nav-icon"
        disabled={false}
        eventKey="facebook"
      >
        <img
          alt="facebook"
          src="fb_logo.svg"
          style={
            Object {
              "height": "21px",
              "width": "20px",
            }
          }
        />
      </NavItem>
      <NavItem
        active={false}
        className="nav-icon"
        disabled={false}
        eventKey="twitter"
      >
        <img
          alt="twitter"
          src="twitter-silhouette.svg"
          style={
            Object {
              "height": "21px",
              "width": "20px",
            }
          }
        />
      </NavItem>
      <NavItem
        active={false}
        className="nav-icon"
        disabled={false}
        eventKey="linkedin"
      >
        <img
          alt="linkedin"
          src="linkedin-logo.svg"
          style={
            Object {
              "height": "21px",
              "width": "20px",
            }
          }
        />
      </NavItem>
      <NavItem
        active={false}
        className="nav-icon"
        disabled={false}
        eventKey="email"
      >
        <img
          alt="email"
          src="email-icon.svg"
          style={
            Object {
              "height": "21px",
              "width": "20px",
            }
          }
        />
      </NavItem>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
